stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/SendMessageListenerImpl.kt

```diff
@@ -81,17 +93,42 @@ internal class SendMessageListenerImpl(
         if (latestUpdatedMessage.syncStatus == SyncStatus.COMPLETED) {
             return
         }
+        val isPermanentError = error.isPermanent()
+        val isMessageModerationFailed = error is ChatNetworkError &&
+            error.streamCode == ChatErrorCode.MESSAGE_MODERATION_FAILED.code
+        StreamLog.w(TAG) {
+            "[handleSendMessageFail] isPermanentError: $isPermanentError" +
+                ", isMessageModerationFailed: $isMessageModerationFailed"
+        }
         message.copy(
-            syncStatus = if (error.isPermanent()) {
+            syncStatus = if (isPermanentError) {
                 SyncStatus.FAILED_PERMANENTLY
             } else {
                 SyncStatus.SYNC_NEEDED
             },
+            syncDescription = error.toSyncDescription(),
             updatedLocallyAt = Date(),
         )
             .also {
                 repos.insertMessage(it)
                 channel.upsertMessage(it)
             }
     }
+
+    private fun ChatError.toSyncDescription(): MessageSyncDescription? {
+        return when (this is ChatNetworkError) {
```

Comment:
Do we need these when statements or can we maybe simplify this?

Something like

```
val networkError = this as? ChatNetworkError ?: return null

// rest of the mapping
```